Implements: The styling of the Blaze Campaigns Card

AjeshRPai opened the pull request:
Part of 
#18521 
#18525 

## Description 
This PR implements the styling of the campaigns dashboard card. 

This PR 
⊕ Adds the UI of the Campaigns Card in Jetpack compose 
⊕ Adds the logic to show the correct status and the label in light/dark mode 
⊕ Adds the logic to show the featured image and title(long title is also taken care of)
⊕ Adds the logic to show the campaign stats if the campaign is **Active|Completed**

## Note
⊘ Does not add the logic to handle tracking or click of the button/card - will be implemented in a future PR

## Campaigns Card 
| Active and Long title(Light) | Active and Long Title(Dark) |  Canceled Campaign Light |  Canceled Campaign Night | 
|---|---|---|---|
| ![campaigns_light](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/0be0c901-e232-457e-8339-8c951130f812)  |  ![campaigns_night](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/e7406ac9-f7cc-48f8-8f1a-a73d6fc7fdc0) |  ![campaign_cancelled_light](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/33927d8b-f0e3-4234-b29f-a29fccd4381b) |   ![cancelled_night](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/a15e6865-806a-4b88-bb90-9c8df508bda5)| 

## Test 
#### Prerequisite
<details><summary>◐ Toggle the <code>blaze_manage_campaign</code> flag</summary>

1. Go to `Me` → `Debug settings`
2. Scroll to the `Features in development` section
3. Tap on the item corresponding to the flag
4. Tap `RESTART THE APP` button
<img width="501" alt="Screenshot 2023-04-27 at 10 59 09 AM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/d219ce2b-2cf2-4641-9f8c-546a45c5e182.png">

</details>

1. Login to the app with an account having a campaign(Active|Rejected|Finished|Completed|Canceled)
2. Go to the Dashboard 
3. Verify that the Recent campaign is shown on the dashboard card
4. Verify the title, featured image url, status and stats are correct
5. Verify that the dashboard card design is as per the expectations in the Figma 
6. Verify that long title of the campaign is handled(ellipsis after two lines)

## Merge Instructions 
- [x] Make sure that https://github.com/wordpress-mobile/WordPress-Android/pull/18686 is merged 
- [x] There are some test failures in the base PR- https://github.com/wordpress-mobile/WordPress-Android/pull/18686 because of which the CI is failing on this branch, it should be fixed before the merge
- [x] Rebase with trunk if needed
- [ ] Merge as normal

## Regression Notes
1. Potential unintended areas of impact
- Campaigns card is not shown as required 
- Details in Campaigns card is incorrect

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and Added Unit tests 

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

AjeshRPai committed:
+ Adds: a composable for dashboard card component

AjeshRPai committed:
* Moves: DashboardCard to compose package

AjeshRPai committed:
+ Adds: material3 depenendancy for material 3 themes

AjeshRPai committed:
+ Adds: styles for text views in dashboard card

AjeshRPai committed:
* Updates: styling of the dashboard card

AjeshRPai committed:
+ Adds: String resources for campaign status

AjeshRPai committed:
+ Adds: the string resource for canceled campaign status

AjeshRPai committed:
+ Adds: enum class for holding UI elements for status of campaign

+ Adds: enum class - CampaignStatus
+ Adds: the color resource helper for getting text color of status
+ Adds: the color resource helper for getting background color of
status text

AjeshRPai committed:
* Updates: the type of campaign status to CampaignStatus to hold data

* Updates: the data type to hold more ui data about the campaign status
and for easier lookup of resources related to status

AjeshRPai committed:
* Updates: the Blaze Campaigns Card View

+ Adds: BlazeStatusLabel compose view for rendering campaign status
+ Adds: the check for darkmode in Card
* Updates: the BlazeCampaignsCardView to use BlazeStatusLabel

AjeshRPai committed:
* Updates: the container of the campaign status label

AjeshRPai committed:
+ Adds: the logic to show the featured image url of the campaign

* Updates: the BlazeCampaignsCardView to show the featured image url
+ Adds: the logic to show the title when there is no featured image
+ Adds: the logic to constraint the title to two lines

AjeshRPai committed:
* Updates: the top padding of the blaze campaign status

AjeshRPai committed:
+ Adds: the dashboard styles for stats in campaigns card

AjeshRPai committed:
* Renames: the member variables for readability

AjeshRPai committed:
* Renames: the member variables for readability in tests

AjeshRPai committed:
+[WIP] Adds: the compose view for campaign stats

+ Adds: the Campaign Stat view and the logic for rendering it

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
 
 @Composable
 @Suppress("FunctionName")
 fun BlazeCampaignsCardView(
```
💡 Suggestion: I would avoid using the suffix `View` for Composables as it might be confused with View-based UIs/components, which usually have that suffix (e.g.: Text**View**, Image**View**, etc). Also, since we are using both XMLs (view-based) and Compose in this codebase I would reserve the `View` suffix for View-wrappers around Composables if needed.

You can find an example here:
I created a more generic `TrainOfIcons` Composable **here** that I needed to use for avatars in XML-based layouts, so I created a `TrainOfAvatarsView` that extends `AbstractComposeView` and renders the `TrainOfIcons` [internally](https://github.com/wordpress-mobile/WordPress-Android/blob/108e3edfd01ecd1c65a3b69c5124c37857967ae4/WordPress/src/main/java/org/wordpress/android/ui/compose/views/TrainOfAvatarsView.kt#L91-L102).

---

Also, I see there's already a `BlazeCampaignsCard` class being used for the model, and TBH I still didn't find a naming system that I like, but I would call the model `BlazeCampaignsCardModel` so I could name the Composable `BlazeCampaignsCard`, which is what [I did for the `TrainOfIconsModel`](https://github.com/wordpress-mobile/WordPress-Android/blob/6c9a94832f2ef36a256fbaf5d66b426abdda5eef/WordPress/src/main/java/org/wordpress/android/ui/compose/components/TrainOfIcons.kt#L34-L37) btw. 😅

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
     modifier: Modifier = Modifier,
-    blazeCampaignCard: BlazeCampaignsCard
+    blazeCampaignCard: BlazeCampaignsCard,
+    isInDarkMode: Boolean = isSystemInDarkTheme()
```
⚠️ The order of the arguments seems swapped, `modifier` should be the first optional parameter in the Composable function, as noted in the [documentation](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier#:~:text=It%20should%20appear%20as%20the%20first%20optional%20parameter%20in%20the%20parameter%20list%3B%20after%20all%20required%20parameters%20(except%20for%20trailing%20lambda%20parameters)%20but%20before%20any%20other%20parameters%20with%20default%20values.):
> It should appear as the first optional parameter in the parameter list; after all required parameters (except for trailing lambda parameters) but before any other parameters with default values.

thomashorta replied:
💡  There's no need to pass `isInDarkMode` as a parameter. Since we are using a `MaterialTheme` root parent for everything we can get that information easily from anywhere using the `LocalColor` composition local, like so:
```kotlin
val isDarkMode = !MaterialTheme.colors.isLight
```

Bear in mind that this `isLight` corresponds to Material `Colors` instances created with `lightColors` and `darkColors` but currently, we are [using them as expected](https://github.com/wordpress-mobile/WordPress-Android/blob/51f154c92574e6f5282901f123cfa6d48e158b16/WordPress/src/main/java/org/wordpress/android/ui/compose/theme/JetpackColors.kt#L10-L39) (`lightColors` for Light mode and `darkColors` for Dark mode), so this should be fine.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
```diff
@@ -0,0 +1,43 @@
+    Card(
+        modifier = modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
```
💡 I don't think `wrapContentHeight` is needed here. I usually only use it when required (the UI doesn't look correct without it) as it might override incoming minimum measurement restrictions from parent Composables and might cause the component to be less reusable.

I would also not add the `fillMaxWidth()` modifier here, since it looks like the idea of `DashboardCard` is to be reusable, so the decision about layout sizes would better be made by other components that use this one, if that makes sense.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
```diff
@@ -0,0 +1,43 @@
+        modifier = modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .padding(1.dp),
```
❓ Why do we need this `1dp` padding here? I couldn't figure out if it's actually needed.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
```diff
@@ -0,0 +1,43 @@
+                .fillMaxHeight()
+        ) {
+            content()
+        }
```
⚠️ I don't think the `Column` should be part of `DashboardCard`. What if I want to implement a Dashboard card that contains only a row or even a single Composable? I think you can remove this column and replace with just the `content` composable call.

```suggestion
        content()
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
```diff
@@ -0,0 +1,43 @@
+            .wrapContentHeight()
+            .padding(1.dp),
+        border = BorderStroke(
+            width = dimensionResource(id = R.dimen.unelevated_card_stroke_width),
```
❓ Is there any particular reason to be using this resource here instead of `1dp`? 

I believe it could be valid to use if, for instance, we want to make sure this matches that dimension that's used in the view-based layouts, so if that changes the Compose counterpart changes as well, just wanted to confirm this is the reason, or something else.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
💡 I'm not sure about the path for this file, as this `DashboardCard` is very specific to the dashboard and not a multipurpose component like the others in the `ui/compose/components/ package. 

Though I actually see the value of it being an unelevated card, especially if the changes I suggested are applied, so maybe renaming it to `UnelevatedCard` would be better, while keeping it here in this package. What do you think?

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+@Composable
+fun largeText() = MaterialTheme.typography.headlineMedium.copy(
+    color = colorResource(R.color.material_on_surface_emphasis_high_type)
+)
```
💡 I liked that the typography styles for dashboard cards are grouped here, but I think it would be good that they were inside an `object`, for better separation, especially because they have very generic names. Here's a suggestion:

Rename the file and create an `object DashboardCardTypography` (to be more in line with how Compose/Material calls this) and put the styles inside it. 

An optional suggestion (more of a nitpick I think) is changing these functions to be properties without a backing field, like this:
```kotlin
val smallTitle: TextStyle
    @Composable
    get() = MaterialTheme.typography.bodyLarge.copy(
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        color = colorResource(R.color.material_on_surface_emphasis_high_type)
    )
```

So callers can do something like `DashboardCardTypography.smallTitle`, which feels closer to how Compose and Compose Material exposes things (though they are using actual properties).

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+fun smallTitle() = MaterialTheme.typography.bodyLarge.copy(
+    fontWeight = FontWeight.Normal,
+    fontSize = 14.sp,
+    color = colorResource(R.color.material_on_surface_emphasis_high_type)
```
⚠️ Compose Material already provides the emphasis alphas (which they call `ContentAlpha`) and `onSurface` color so you could use them directly for getting this color. They also behave correctly in High/Low contrast system options, so they are more recommended to use in this case.
```suggestion
    color = MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.high)
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+fun subTitle() = MaterialTheme.typography.titleMedium.copy(
+    fontWeight = FontWeight.Medium,
+    fontStyle = FontStyle.Normal,
+    color = colorResource(R.color.material_on_surface_emphasis_high_type)
```
⚠️ Same here.
```suggestion
    color = MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.high)
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+fun detailText() = MaterialTheme.typography.bodyLarge.copy(
+    fontWeight = FontWeight.Normal,
+    fontSize = 14.sp,
+    color = colorResource(R.color.material_on_surface_emphasis_medium)
```
⚠️ Same here.
```suggestion
    color = MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.medium)
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+
+@Composable
+fun largeText() = MaterialTheme.typography.headlineMedium.copy(
+    color = colorResource(R.color.material_on_surface_emphasis_high_type)
```
⚠️ Same here.
```suggestion
    color = MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.high)
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/styles/DashboardCardStyle.kt:
```diff
@@ -0,0 +1,41 @@
+fun footerCTA() = MaterialTheme.typography.titleMedium.copy(
+    fontStyle = FontStyle.Normal,
+    color = colorResource(R.color.primary_emphasis_medium_selector)
+)
```
⚠️ This is an interesting style, from what I've seen in the code it is expected that this will be applied to a `Text` that will behave as a button. Using a resource with a selector does not work as `colorResource` does not support `ColorStateList`, so it will just get the default color and not react to changes in defined states (like `pressed` in the case of this resource).

I think what you want to use in that case is the `SecondaryButton` from [here](https://github.com/wordpress-mobile/WordPress-Android/blob/022e4d76755440dca10e13cae580251ee09cfcda/WordPress/src/main/java/org/wordpress/android/ui/compose/components/buttons/SecondaryButton.kt#L25) (for some reason there are several Secondary buttons with same name, be sure to use the one from `ui/compose/components` in a way similar to what I did [here](https://github.com/wordpress-mobile/WordPress-Android/blob/26f817f6a2ab3891bc5108716b2e797bd4e9b133/WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/compose/PrepublishingHomeSocialConnectItem.kt#L77-L83), that is, passing `PaddingValues(0.dp) to the `padding` and `contentPadding`. It is not perfect, and it has a `ripple` instead of a subtle text color change, but should work.

I thought about this issue earlier this week as well, and we might want to create a custom `TextButton` (or modify `SecondaryButton`) to be closer to what we would expect for this kind of button, at least making it similar to what we have in the View-based side of things in our repo, which is usually a text that gets some transparency when clicked. 🤔

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
-                    .fillMaxWidth()
-                    .wrapContentHeight()
-                    .padding(start = 16.dp, top = 8.dp)
+                .wrapContentHeight()
```
💡 There's no need for `wrapContentHeight` here, I suggest removing it.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            featuredImageUrl = blazeCampaignCard.campaign.featuredImageUrl,
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
```
💡 There's no need for `wrapContentHeight` here, I suggest removing it.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+                campaignStats = blazeCampaignCard.campaign.stats,
+                modifier = Modifier
+                    .wrapContentWidth()
                     .wrapContentHeight()
```
💡 There's no need for `wrapContentHeight` or `wrapContentWidth` here unless I am not seeing something, so I suggest removing it.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            textAlign = TextAlign.Start,
+            modifier = modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
```
💡 There's no need for `wrapContentHeight` here, I suggest removing it.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            style = smallTitle(),
+            textAlign = TextAlign.Start,
             modifier = modifier
                 .fillMaxWidth()
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `DashboardCard` on line 50 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M). 

Some basics about Modifiers [here](https://developer.android.com/jetpack/compose/modifiers)].

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/compose/components/card/DashboardCard.kt:
```diff
@@ -0,0 +1,43 @@
+        shape = RoundedCornerShape(10.dp)
+    ) {
+        Column(
+            modifier = modifier
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `Card` on line 23 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M). 

Some basics about Modifiers [here](https://developer.android.com/jetpack/compose/modifiers)].

Note: I talked about removing this `Column` anyway but just adding this as a heads up as it seems all children Composables everywhere in this PR did that mistake.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            text = uiStringText(uiString = blazeCampaignCard.footer.label),
+            style = footerCTA(),
+            textAlign = TextAlign.Start,
+            modifier = modifier
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `DashboardCard` on line 50 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M).

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+fun CampaignTitleThumbnail(campaignTitle: UiString, featuredImageUrl: String?, modifier: Modifier = Modifier) {
+    ConstraintLayout(
+        modifier = modifier
+            .padding(horizontal = 16.dp, vertical = 8.dp)
```
💡 Avoid putting margins that affect the positioning of the component inside the Component itself. Here, for instance, this padding will be around the `ConstraintLayout` (equivalent to an XML layout `margin`) so it would make sense for it to be defined externally when the caller passes a `modifier` in the parameters.

 For example, here it would make more sense for the parent rendering this Composable to position it (using `padding` or `Spacer`s) than for it to simply call it and expect it to have the correct margins. 

Thinking in Compose is a bit different than dealing with Views and Layouts, and I don't know if I am being able to explain it properly, but there is this nice resource about it as well: https://youtu.be/4zf30a34OOA

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+        style = subTitle(),
+        maxLines = 2,
+        overflow = TextOverflow.Ellipsis,
+        modifier = modifier.wrapContentHeight()
```
💡   This `wrapContentHeight` is not really needed here, you can just do:
```
modifier = modifier
```

Also, if it DOES end up being needed, it makes more sense to the parent to pass it in the `modifier` parameter anyway, and not have it already there inside this internal Composable.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+    Row(
+        modifier = modifier
+            .wrapContentHeight()
+            .wrapContentWidth(),
```
💡 I don't think these `wrap` modifiers are needed here. By default, the Row will try to use only the needed size to wrap its children anyway. Also, if you do end up needing to have some of those layout modifiers it would make more sense for them to be passed by the parent of this Composable.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            .wrapContentHeight()
+            .wrapContentWidth(),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.Start
```
🔍 This is the default arrangement for `Row` so it's technically not needed here.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            textAlign = TextAlign.Start,
+            modifier = modifier
+                .wrapContentWidth()
+                .wrapContentHeight()
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `Column` on line 194 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M).

💡 Also, this `Text` likely does not need those `wrap` modifiers as the default behavior for text is to just use the minimum space required (unless modified by external factors, which is not the case here).

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            textAlign = TextAlign.Start,
+            modifier = modifier
+                .wrapContentWidth()
+                .wrapContentHeight()
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `Column` on line 194 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M).

💡 Also, this `Text` likely does not need those `wrap` modifiers as the default behavior for text is to just use the minimum space required (unless modified by external factors, which is not the case here).

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+    Column(
+        modifier = modifier
+            .wrapContentHeight()
+            .wrapContentWidth()
```
💡 You don't seem to need these `wrap` modifiers here. This could just be:
```suggestion
        modifier = modifier
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+        CampaignStat(
+            modifier = modifier
+                .wrapContentHeight()
+                .wrapContentWidth(),
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `Row` on line 167 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M).

💡 Also, this Composable does not need those `wrap` modifiers, so the `modifier` parameter can be omitted altogether.

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+            modifier = modifier
+                .wrapContentHeight()
+                .wrapContentWidth()
+                .padding(end = 8.dp),
```
🛑  You should **not** be using the `modifier` passed as a parameter for children, only for the topmost parent Composable which wraps all children (in this case the `Row` on line 167 is correct). All children that need modifiers to configure layouts need to use the base `Modifier` (capital M).

💡 Also, this Composable does not need those `wrap` modifiers.

❓ I didn't understand the need for this `end` padding. I am guessing this is still WIP, right?

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
+
+@Composable
+fun CampaignTitleThumbnail(campaignTitle: UiString, featuredImageUrl: String?, modifier: Modifier = Modifier) {
+    ConstraintLayout(
```
🔍 This doesn't look like a good use case for [`ConstraintLayout`](https://developer.android.com/jetpack/compose/layouts/constraintlayout) as this layout seems to be a very simple `Row` with a Text and an Image right next to it unless I am missing something.

Here's a draft of how this could look like:
```kotlin
Row(
    modifier = modifier,
    horizontalArrangement = Arrangement.SpaceBetween,
    verticalAlignment = Alignment.CenterVertically,
) {
    CampaignTitle(
        title = uiStringText(uiString = campaignTitle),
        modifier = Modifier.weight(1f, fill = false), // trick to make sure the text is properly measured and doesn't push the image out of the `Row` in case it's too long
    )
    
    if (featuredImageUrl != null) {
        Spacer(Modifier.width(16.dp)) // min space between text and image
        
        FeaturedImage(url = featuredImageUrl)
```

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt:
```diff
@@ -1,93 +1,216 @@
                     .wrapContentHeight()
-                    .padding(start = 16.dp, top = 8.dp)
+                    .background(Color.LightGray)
+                    .padding(start = 16.dp, top = 8.dp, bottom = 8.dp)
```
💡 This `top` padding is not the usual way of doing spacing between items in Compose. It's far more common to use [`Spacer`](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/package-summary#Spacer(androidx.compose.ui.Modifier)) (with `Modifier.width()` for Rows, and `Modifier.height()` for columns) or, if all spacing between items in those containers is equal, you can play with [`Arrangement`](https://developer.android.com/reference/kotlin/androidx/compose/foundation/layout/Arrangement): `horizontalArrangement` (for rows) or `verticalArrangement` (for columns) with `Arrangement.spacedBy()` and `Arrangemement.SpaceBetween`.

The `padding()` modifier is mostly used for internal paddings inside components or around children in container layouts like Row and Column, so the equivalent of View-based `padding` and not `margin`.

As a rule of thumb, you can think about it this way: 
- if you're trying to achieve something that would be a `margin` between components in an XML layout, you probably want a `Spacer` or `Arrangement`.
- if you want to try to achieve what would be a `padding` inside a component in an XML layout or a `margin` **around** your component, you probably want a `padding()` modifier.

Overall that "margin around component" scenario is not very common since that can be done as a `padding` inside the parent container composable, which is usually a Row, Column, or Box, once you get used to thinking in Compose.

thomashorta reviewed (dismissed):
Thanks, @AjeshRPai for the changes, and kudos for getting your hands "dirty" with Compose and getting more familiar with it. 🙇 

I added several comments and suggestions to improve the Compose part of the code, I hope you don't mind. I tried my best to explain some things but I don't know if I was able to make myself 100% clear, so feel free to ping or DM me to talk more about it.

Also, I know you are starting your Compose journey as well so I would recommend looking at some courses from [here](https://developer.android.com/courses/jetpack-compose/course), which are official resources from the Android team and very useful to get started. I personally didn't do all of them (yet) but they helped me a lot to get a good grasp of how to think in Compose and use it, so I hope they can be helpful to you too.

Thanks again!

thomashorta commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeStatusLabel.kt:
```diff
@@ -0,0 +1,52 @@
+fun BlazeStatusLabel(
+    modifier: Modifier = Modifier,
+    status: CampaignStatus,
+    isInDarkMode: Boolean = false
```
💡 This is not needed, you can access `MaterialTheme.colors.isLight`

AjeshRPai committed:
* Updates: Dashboard card styles with better approach

* Updates: the styles to be an object and updates the
 colors to use content alpha
* Updates: the usages of these styles

AjeshRPai committed:
* Renames: the BlazeCampaignCard data class to BlazeCampaignCardModel

AjeshRPai committed:
* Renames: BlazeCampaignsCardView.kt to BlazeCampaignsCard

* Updates: the parameter from blazeCampaignCard to
blazeCampaignCardModel

AjeshRPai committed:
-> Moves: Modifier to be the first optional parameter

AjeshRPai committed:
- Removes: isSystemInDarkTheme parameter from Compose views

* Replaces: with !MaterialTheme.colors.isLight check at the usage

AjeshRPai committed:
* Updates: the usage of Modifier in child compose views

* Updates: the usage of the modifier in the child views, modifier
 should be used only in direct child and not in nested child

AjeshRPai committed:
- Removes: the redundant wrap content size modifiers in views

AjeshRPai committed:
- Removes: redundant horizontalArrangement parameter

AjeshRPai committed:
* Updates: the rendering logic of DashboardCard

- Removes: the column child view from DashboardCard
- Removes: the redudant modifiers
* Updates: the width and color resource values
* Updates: the BlazeCampaignsCard to use column and
Arrangemenet.spacedBy for spacing between elements

AjeshRPai committed:
* Updates: the logic of CampaignTitleThumbnail compose view

- Removes: the constraint layout with Row layout
- Removes: the extra compose view

AjeshRPai committed:
* Updates: variable naming

AjeshRPai committed:
* Removes: redundant modifier

AjeshRPai committed:
* Renames: DashboardCard.kt to UnelevatedCard.kt

AjeshRPai committed:
* Updates: the dashboard card logic to correct padding between elements

+ Adds: divider to the dashboard card
+ Adds: Column parent composables to fix the padding of different
elements

AjeshRPai committed:
* Changes: the order of the parameters

AjeshRPai committed:
Merge branch 'issue/18522-blaze-manage-campaigns-implement-the-empty-campaigns-logic-for-dashboard-card' into issue/styling-of-the-dashboard-card

AjeshRPai committed:
Adds: the logic to show/hide campaign stats

AjeshRPai committed:
Adds: Spacer when campaign stats is not shown

AjeshRPai committed:
Fixes: lint error

AjeshRPai commented:
Hey @thomashorta 

Thanks for reviewing and giving me the feedback about the compose code, its been really helpful. 

I have made the necessary changes as per your suggestions. Also thanks for explaining all the changes required in such detail and with patience. You 🪨 .

AjeshRPai committed:
+ Adds: string resources for campaign stats header

AjeshRPai committed:
Merge branch 'trunk' into issue/styling-of-the-dashboard-card

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/BlazeCampaignsCardView.kt

AjeshRPai committed:
+ Adds: Scheduled Campaign status

+ Adds: the string resource for Scheduled campaign status
+ Updates: the In moderation status value and updates the rest of the
values to remove the first capital letter

AjeshRPai committed:
* Fixes: Incorrect unit tests

AjeshRPai committed:
- Removes: Redundant class

AjeshRPai committed:
* Updates: Scheduled status label and background colors

AjeshRPai committed:
* Fixes: typo and re-arranges the string resources

zwarm commented on WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/blaze/CampaignStatus.kt:
```diff
@@ -28,7 +29,7 @@ enum class CampaignStatus(val status: String, @StringRes val stringResource: Int
     fun textColor(isInDarkMode: Boolean): Color {
```
@AjeshRPai - ❓ why not a color resource instead of hardcoding?

zwarm reviewed (approved):
@AjeshRPai - I left one question, but it's not a blocker. I used the `testblazecampaign` account to see most of the statuses; however I used the branch + fluxC locally and made direct changes to the response for others. I also tested landscape/portrait + large/small font size nad bold text. Overall everything looks great. I know you are waiting for design review, so I'll leave my approval and no merge.

@osullivanchris - The 'canceled' red/on red looks like it's glowing when in dark mode. Could just be my eyes, but wanted to call it out.

@thomashorta - thanks for the fantastic job reviewing all the compose code. I learned so much 👍

AjeshRPai committed:
* Updates: Capitalizes the string resource as per design

osullivanchris reviewed (approved):
Working as expected for me

> @osullivanchris - The 'canceled' red/on red looks like it's glowing when in dark mode. Could just be my eyes, but wanted to call it out.

This is fair. Not just the red labels, but they all are super vibrant in dark mode like toxic waste in a cartoon hah. If we want to tweak, the 10 rather than 30 values on dark mode could soften things (e.g. red-10 and green-10 rather than red-30 and green-30 for the text). [Values here](https://color-studio.blog/).
